src/plugins/skills/fishing/fishing-task.ts

```diff
@@ -0,0 +1,93 @@
+import { Skill } from '@engine/world/actor/skills';
+import {  Player, Npc } from '@engine/world/actor';
+import { soundIds } from '@engine/world/config/sound-ids';
+import { findItem, findNpc, findObject } from '@engine/config/config-handler';
+import { activeWorld } from '@engine/world';
+import { ActorActorInteractionTask } from '@engine/task/impl';
+import { LandscapeObject } from '@runejs/filestore';
+import { logger } from '@runejs/common';
+import { IHarvestable } from '@engine/world/config';
+import { canInitiateHarvest } from '@engine/world/skill-util/harvest-skill';
+import { randomBetween } from '@engine/util';
+
+
+class FishingTask extends ActorActorInteractionTask<Player, Npc>{
+
+    private elapsedTicks = 0;
+
+    constructor(
+        player: Player,
+        fishingSpot: Npc,
+    ) {
+        super(
+            player,
+            fishingSpot,
+            1,
+            1
+        );
+
+
+        if (!fishingSpot) {
+            this.stop();
+            return;
+        }
+
+    }
+    public execute(): void {
+        super.execute()
+
+        if (!this.isActive || !this.otherActor) {
+            return;
+        }
+
+        // store the tick count before incrementing so we don't need to keep track of it in all the separate branches
+        const taskIteration = this.elapsedTicks++;
+
+        // TODO wire this up into fishing spot config
+        const fishingSpotInfo: Pick<IHarvestable, 'itemId' | 'level'> = {
+            itemId: 335,
+
+            // TODO change this too
+            level: 1
+        }
+
+        const tool = canInitiateHarvest(this.actor, fishingSpotInfo, Skill.FISHING);
+
+        if (!tool) {
+            this.stop();
+            return;
+        }
+        if(taskIteration === 0) {
+            this.actor.sendMessage('You swing your axe at the tree.');
+            this.actor.face(this.otherActor);
+            this.actor.playAnimation(tool.animation);
+            return;
+        }
+        const roll = randomBetween(1, 256)
+        if(roll > 200){
```

Comment:
we should move this into a separate file (see woodcutting, which has its own `chance` file) though also some research is needed on the OSRS wiki to see how fishing chances are handled